### PR TITLE
Add basic planner

### PR DESCRIPTION
**Pull request by jcnorby**
Okay, this is the first pass at basic global body and local footstep planners. This is a pretty big commit so let me know if anything needs updating.

### CODE REVIEW

**Comment by jcnorby** [comment]
There's a good chance I missed some dependencies so we'll see what travis has to say about them, but any reviews on the content itself would be helpful.

File: global_body_planner/include/global_body_planner/functions.h [hunk 1]
```diff
@@ -0,0 +1,122 @@
+#ifndef FUNCTIONS_H
+#define FUNCTIONS_H
```
Comment (nikolaif399) [inline global_body_planner/include/global_body_planner/functions.h:2]: Is there a way to namespace these functions and #define's? We should also think about renaming it to something more descriptive?.

File: global_body_planner/include/global_body_planner/rrt.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#define RRT_H
+
+#include "global_body_planner/planner_class.h"
+#include <chrono>
+
+#define TRAPPED 0
+#define ADVANCED 1
+#define REACHED 2
+
+class RRTClass
```
Comment (nikolaif399) [inline global_body_planner/include/global_body_planner/rrt.h:11]: Should this be a child of PlannerClass?

File: global_body_planner/include/global_body_planner/rrt.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        virtual int extend(PlannerClass &T, State s, Ground& ground, int direction);
+        std::vector<int> pathFromStart(PlannerClass &T, int s);
+        void printPath(PlannerClass &T, std::vector<int> path);
+        void buildRRT(Ground &ground, State s_start, State s_goal, std::vector<State> &state_sequence, std::vector<Action> &action_sequence);
+        void getStatistics(double &plan_time, int &success_var, int &vertices_generated, double &time_to_first_solve, std::vector<double> &cost_vector, std::vector<double> &cost_vector_times, double& path_duration);
+
+        bool newConfig(State s, State s_near, State &s_new, Action &a_new, Ground &ground, int direction);
+
+        std::vector<State> getStateSequence(PlannerClass &T, std::vector<int> path);
+        std::vector<Action> getActionSequence(PlannerClass &T, std::vector<int> path);
```
Comment (nikolaif399) [inline global_body_planner/include/global_body_planner/rrt.h:29]: Why the need to accept a PlannerClass obj as an argument here? Could this be solved by subclassing PlannerClass?

**Review by nikolaif399 — APPROVED** [review]
Couple changes to be made, but mostly semantic/aesthetic stuff that can be taken care of along the line. Pulling in to master now.

**Pull request merged**